src/tests/ftest/util/general_utils.py

```diff
@@ -39,6 +40,122 @@ class DaosTestError(Exception):
 
 
+class SimpleProfiler(object):
+    """
+    Simple profiler that counts the number of times a function is called
+    and measure its execution time.
+    """
+
+    def __init__(self):
+        self._stats = {}
+        self._logger = None
+
+    def clean(self):
+        """
+        Clean the metrics collect so far.
+        """
+        self._stats = {}
+
+    def run(self, fn, tag, *args, **kwargs):
+        """
+        Run a function and update its stats.
+
+        Parameters:
+            fn (function): Function to be executed
+            args  (tuple): Argument list
+            kwargs (dict): Keyworded, variable-length argument list
+        """
+        self._log("Running function: {0}()".format(fn.__name__))
+
+        start_time = time.time()
+
+        ret = fn(*args, **kwargs)
+
+        end_time = time.time()
+        elapsed_time = end_time - start_time
+        self._log(
+            "Execution time: {0}".format(
+                _pretty_time(elapsed_time)))
+
+        if tag not in self._stats:
+            self._stats[tag] = [0, []]
+
+        self._stats[tag][0] += 1
+        self._stats[tag][1].append(elapsed_time)
+
+        return ret
+
+    def get_stat(self, tag):
+        """
+        Retrieves the stats of a function.
+
+        Parameters:
+            tag (str): Tag to be query
+
+        Return:
+            max, min, avg (tuple): A tuple with the slowest, fastest and
+            average execution times.
+        """
+        data = self._stats.get(tag, [0, []])
+
+        return _calculate_metrics(data)
+
+    def set_logger(self, fn):
+        """
+        Set the function that will be used to print the elapsed time on each
+        function call. If this value is not set, the profiling will be
+        performed silently.
+
+        Parameters:
+            fn (function): Function to be used for logging.
+        """
+        self._logger = fn
+
+    def print_stats(self):
+        """
+        Prints all the stats collected so far. If the logger has not been set,
+        the stats will be printed by using the built-in print function.
+        """
+        self._pmsg("{0:20} {1:5} {2:10} {3:10} {4:10}".format(
+            "Function Tag", "Hits", "Max", "Min", "Average"))
+
+        for fname, data in self._stats.items():
+            max_time, min_time, avg_time = _calculate_metrics(data)
+            self._pmsg(
+                "{0:20} {1:5} {2:10} {3:10} {4:10}".format(
+                    fname,
+                    data[0],
+                    _pretty_time(max_time),
+                    _pretty_time(min_time),
+                    _pretty_time(avg_time)))
+
+    def _log(self, msg):
+        if self._logger:
+            self._logger(msg)
+
+    def _pmsg(self, msg):
+        if self._logger:
+            self._log(msg)
+        else:
+            print(msg)
+
+
+def _pretty_time(ftime):
+    return time.strftime("%H:%M:%S", time.gmtime(ftime))
+
+
+def _calculate_metrics(data):
```

Comment:
What kind of object is data? It would be great to have an example of it when using this function